From b6fbf676f43505a2cc9e7a59801642008c71b766 Mon Sep 17 00:00:00 2001
From: Jiafei Pan <Jiafei.Pan@nxp.com>
Date: Fri, 27 Oct 2023 18:35:37 +0800
Subject: [PATCH 14/23] arm64: dts: imx8mm/p: optimize gen-sw-mailbox ack for
 rpmsg

Now for the mailbox message sent from RTOS to linux, Linux will sent
back a ACK interrupt to RTOS when Linux received this message, but
RTOS no any callback funtion to be handled in this interrupt handler
except only update mailbox status register, so remove this ACK interrupt
to get a better performance.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>
---
 arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi  | 6 +++---
 arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi b/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi
index 77f0d92ceb69..a3b0dbf563e8 100644
--- a/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi
@@ -29,9 +29,9 @@ rpmsg-ca53 {
 		dma-coherent;
 		memory-region = <&vdevbuffer_ca53>;
 		mbox-names = "tx", "rx", "rxdb";
-		mboxes = <&gen_sw_mbox 0 0
-			  &gen_sw_mbox 1 0
-			  &gen_sw_mbox 2 0>;
+		mboxes = <&gen_sw_mbox 0 0 1	/* Tx channel with ACK */
+			  &gen_sw_mbox 1 0 0	/* Rx channel without ACK */
+			  &gen_sw_mbox 2 0 1>;	/* RXDB channel with ACK */
 		vdev-nums = <1>;
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
index 1bb2a8602379..32e127a6d203 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
@@ -29,9 +29,9 @@ rpmsg-ca53 {
 		dma-coherent;
 		memory-region = <&vdevbuffer_ca53>;
 		mbox-names = "tx", "rx", "rxdb";
-		mboxes = <&gen_sw_mbox 0 0
-			  &gen_sw_mbox 1 0
-			  &gen_sw_mbox 2 0>;
+		mboxes = <&gen_sw_mbox 0 0 1	/* Tx channel with ACK */
+			  &gen_sw_mbox 1 0 0	/* Rx channel without ACK */
+			  &gen_sw_mbox 2 0 1>;	/* RXDB channel with ACK */
 		vdev-nums = <1>;
 	};
 };
-- 
2.34.1

